### DISCUSSION TITLE
What is the most efficient ? One big query or multiple small queries

### DISCUSSION THREAD

**Discussion by yannisoo** [isar/isar#334] [Help] [answered]
I'm working on a cross plateform app that has a large set of data. I'm currently migrating to Isar and wondering what approach I should take. I have multiple collections which I'm getting from different endpoints and I'm wondering if I should await them all or if I should write in each collection as soon as I get my data. Thanks in advance!

**Comment by simc** [comment, marked as answer]
Hey, I'm not quite sure what you mean. Could you maybe give a small example?

**Reply by yannisoo** [reply]
The goal is to store multiple datasets which I'm getting from seperates endpoints. What I'm doing right now is fetching every Endpoints and typing them as I get them. Then putting all of my typed results in an array which I store in isar with a single write Transaction. I was wondering if I could store them as I type them but I think I may get a concurrential write issue. (Some of these datasets are pretty expensive with about 40 000 entries total and they are quite complex objects as well.

![image](https://user-images.githubusercontent.com/36623256/164887600-c4cdaa37-c27e-4059-b5d2-daba2e13807e.png)

**Reply by simc** [reply]
Okay I understand now. Generally it is a good idea to use as few write transactions as possible but they are not crazy expensive so I think it is fine either way.

Write transactions block each other. They won't block your UI isolate but the future of a write transaction only completes once the previous write transaction is finished. But you don't have to be careful about concurrent write, Isar takes care of that :+1:

**Reply by yannisoo** [reply]
That's great thanks. (by the way isar rocks)
